### PR TITLE
Fix bug of capsule position with vtk<9.3 using non-default parameters

### LINKED ISSUE #6118: Bug with non-default center and direction for capsule with VTK<9.3
### Describe the bug, what's wrong, and what you expected.

I have encountered a bug with the Capsule shape with VTK<9.3 and a non-default center and direction.

The following code is used to generate the figures, I use a cylinder to compare the correct position and orientation.

### Steps to reproduce the bug.

```python
import pyvista as pv

center = (1, 1, 1)
direction = (1, 1, 0)

capsule = pv.Capsule(center=center, direction=direction)
cylinder = pv.Cylinder(center=center, direction=direction)

pl = pv.Plotter()
pl.add_mesh(capsule, color="r", opacity=0.5)
pl.add_mesh(cylinder, color="b", opacity=0.5)
pl.add_axes()
pl.view_xy()
pl.show()
```

### System Information

```shell
--------------------------------------------------------------------------------      
  Date: Fri May 24 18:45:10 2024 Romance Summer Time

                OS : Windows
            CPU(s) : 32
           Machine : AMD64
      Architecture : 64bit
       Environment : Python
        GPU Vendor : NVIDIA Corporation
      GPU Renderer : NVIDIA GeForce RTX 4090/PCIe/SSE2
       GPU Version : 4.5.0 NVIDIA 551.78
  MathText Support : False

  Python 3.10.11 (tags/v3.10.11:7d4cc5a, Apr  5 2023, 00:38:17) [MSC v.1929 64        
  bit (AMD64)]

           pyvista : 0.44.dev0
               vtk : 9.3.0
             numpy : 1.26.4
        matplotlib : 3.9.0
            scooby : 0.9.2
             pooch : 1.8.1
            pillow : 10.3.0
           imageio : 2.34.1
         pyvistaqt : 0.11.0
             PyQt5 : Version unknown
             scipy : 1.12.0
              tqdm : 4.66.2
            meshio : 5.3.5
--------------------------------------------------------------------------------
```

### Screenshots

| VTK 9.2.6 |  VTK 9.3.0 |
:-------------------------:|:-------------------------:
  |

### PR DESCRIPTION
**Pull request by kmarchais**
### Overview

Fixes #6118 

Using the default center and direction (removing them) for the `CapsuleSource` with 9.0 <= VTK < 9.3 allows to fix the bug as the required center and orientation are used in the `translate` method.

The test was failing with VTK < 9.3 before fixing the bug, it passes after the fix commit.

### CODE REVIEW

File: tests/core/test_geometric_objects.py [hunk 1]
```diff
@@ -410,6 +410,16 @@ def test_capsule():
     assert np.any(capsule.faces)
 
 
+@pytest.mark.paramatrize('center', [(4, 5, 6), (1, 1, 1)])
+@pytest.mark.paramatrize('direction', [(0, 1, -1), (1, 1, 0)])
+def test_capsule_center_corresponds_with_cylinder_center(center, direction):
```
Comment (tkoyama010) [inline tests/core/test_geometric_objects.py]: Briefly name the test and add the URL of this PR for explanation.
```suggestion
# https://github.com/pyvista/pyvista/pull/6119
@pytest.mark.paramatrize('center', [(4, 5, 6), (1, 1, 1)])
@pytest.mark.paramatrize('direction', [(0, 1, -1), (1, 1, 0)])
def test_capsule_center(center, direction):
```

File: tests/core/test_geometric_objects.py [hunk 2]
```diff
@@ -433,6 +433,17 @@ def test_capsule():
...
 
 
+# https://github.com/pyvista/pyvista/pull/6119
+@pytest.mark.parametrize('center', [(4, 5, 6), (1, 1, 1)])
+@pytest.mark.parametrize('direction', [(0, 1, -1), (1, 1, 0)])
+def test_capsule_center(center, direction):
+    capsule = pv.Capsule(center, direction)
+    assert np.allclose(capsule.center, center)
+    cylinder = pv.Cylinder(center, direction)
+    assert np.allclose(cylinder.center, center)
```
Comment (tkoyama010) [inline tests/core/test_geometric_objects.py:443]:
```suggestion
    assert np.allclose(cylinder.direction, direction)
```

**Review by tkoyama010 — CHANGES_REQUESTED** [review]

File: tests/core/test_geometric_objects.py [hunk 3]
```diff
@@ -433,6 +433,17 @@ def test_capsule():
     assert np.any(capsule.faces)
 
 
+# https://github.com/pyvista/pyvista/pull/6119
+@pytest.mark.parametrize('center', [(4, 5, 6), (1, 1, 1)])
+@pytest.mark.parametrize('direction', [(0, 1, -1), (1, 1, 0)])
+def test_capsule_center(center, direction):
+    capsule = pv.Capsule(center, direction)
+    assert np.allclose(capsule.center, center)
```
[the remark this answers is not in the document]
Comment (tkoyama010) [reply tests/core/test_geometric_objects.py:441]: Thanks, I change my suggestion.
```suggestion
```

File: tests/core/test_geometric_objects.py (hunk 2 above)
Comment (tkoyama010) [reply tests/core/test_geometric_objects.py:443]:
```suggestion
```

**Comment by tkoyama010** [comment]
pre-commit.ci autofix

**Comment by tkoyama010** [comment]
pre-commit.ci autofix

**Review by tkoyama010 — APPROVED** [review]
LGTM

**Comment by kmarchais** [comment]
Regarding modifying the `CapsuleSource` instead of the `Capsule` function, it appears that the `CapsuleSource` is simply utilizing VTK's implementation. I'm not sure whether this fix should be applied to VTK's `CapsuleSource` or pyvista's.

For now I have left the fix in the `Capsule` function.

**Comment by tkoyama010** [comment]
> Regarding modifying the `CapsuleSource` instead of the `Capsule` function, it appears that the `CapsuleSource` is simply utilizing VTK's implementation. I'm not sure whether this fix should be applied to VTK's `CapsuleSource` or pyvista's.
> 
> For now I have left the fix in the `Capsule` function.

I don't see it as a problem as this issue will be resolved once vtk<9.3 support is discontinued. If a user asks a question, let's think about it again.

**Comment by kmarchais** [comment]
Ok thank you both for the help!

**Review by user27182 — COMMENTED** [review]
LGTM thanks!

**Pull request merged**